programs/drift-competitions/src/instructions/request_randomness.rs

```diff
@@ -0,0 +1,82 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{Token};
+use switchboard_solana::prelude::*;
+use crate::state::Competition;
+
+pub fn request_randomness<'info>(
+    ctx: Context<'_, '_, '_, 'info, RequestRandomness<'info>>,
+) -> Result<()> {
+    // Create the Switchboard request account.
+    let request_init_and_trigger_ctx = FunctionRequestTrigger {
+        request: ctx.accounts.switchboard_request.clone(),
+        function: ctx.accounts.switchboard_function.to_account_info(),
+        authority: ctx.accounts.sponsor.to_account_info(),
+        escrow: ctx.accounts.switchboard_request_escrow.clone(),
+        state: ctx.accounts.switchboard_state.to_account_info(),
+        attestation_queue: ctx.accounts.switchboard_attestation_queue.to_account_info(),
+        payer: ctx.accounts.sponsor.to_account_info(),
+        system_program: ctx.accounts.system_program.to_account_info(),
+        token_program: ctx.accounts.token_program.to_account_info(),
+    };
+
+    msg!("function {:?}", ctx.accounts.switchboard_function.key());
+
+    request_init_and_trigger_ctx.invoke(
+        ctx.accounts.switchboard.clone(),
+        // bounty - optional fee to reward oracles for priority processing
+        // default: 0 lamports
+        Some(1),
+        None,
+        Some(512),
+    )?;
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+pub struct RequestRandomness<'info> {
+    // COMPETITION ACCOUNTS
+    #[account(mut)]
+    pub sponsor: Signer<'info>,
```

Comment:
so only the sponsor can request randomness?

Reply:
nah i changed it